lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -1951,19 +1951,35 @@ reapplyFunctionConversion(SILValue newFunc, SILValue oldFunc,
         builder.createDeallocStack(loc, alloc);
     };
     for (auto arg : pai->getArguments()) {
-      // Retain the argument since it's to be owned by the newly created
+      // Retain the argument if it's to be owned by the newly created
       // closure.
+      // Objects are to be retained.
       if (arg->getType().isObject()) {
         builder.createRetainValue(loc, arg, builder.getDefaultAtomicity());
         newArgs.push_back(arg);
```

Comment:
Minor: how about using early `continue` after each call to `newArgs.push_back` to reduce indentation?

Reply:
Done. PTAL.

Reply:
Looks great!